tools/halide_benchmark.h

```diff
@@ -20,19 +59,69 @@ namespace Tools {
 // for real-world use. For now, callers using this to benchmark GPU
 // code should measure with extreme caution.
 
-template <typename F>
-double benchmark(int samples, int iterations, F op) {
-    double best = std::numeric_limits<double>::infinity();
-    for (int i = 0; i < samples; i++) {
-        auto t1 = std::chrono::high_resolution_clock::now();
-        for (int j = 0; j < iterations; j++) {
-            op();
+struct BenchmarkConfig {
+    // Terminate when the relative difference between the best runtime
+    // seen and the third-best runtime seen is no more than
+    // this. Controls accuracy. The closer to zero this gets the more
+    // reliable the answer, but the longer it may take to run.
+    double accuracy {0.03};
```

Comment:
(Unrelated style nit, but I thought that preferred brace-init syntax elided the space between identifier and brace... at least, that's how I've been doing it.)